File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp = x;
        int revNum = 0;
        while(x > 0){
            int lastDigit = x % 10;
            revNum = revNum * 10 + lastDigit;
            x = x/10;
        }

        return temp==revNum;
    }
}

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long long digit = x;
        long long rev = 0;

        if(x<0){
            return false;
        }

        while(digit!=0){
            rev = rev*10LL + digit%10LL;
            digit = digit/10;
        }

        return x==rev;
    }
};